com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -698,6 +726,22 @@ public void AddReward(float increment)
             m_CumulativeReward += increment;
         }
 
+        public void SetTeamReward(float reward)
```

Comment:
Is there anything preventing the user from adding team rewards manually? How will these be handled if there is no TeamManager for the Agent ?

Reply:
What about making this method `internal` and only calling it from BaseTeamManager methods instead?

Reply:
> How will these be handled if there is no TeamManager for the Agent ?

Team reward will be processed (sum, mean, etc) across the team in python so if it has no TeamManager it will be processed across a team of itself. It's not meaningful in that case but it will run.

Making it internal.